### PR TITLE
adds missing fields to wallet transaction RPCs

### PR DESCRIPTION
**Pull request by hughy**
## Summary

adds transaction fields needed for developers to determine transaction status independently of the node

- adds blockHash and blockSequence to wallet/getAccountTransactions
- adds submittedSequence to responses for both wallet/getAccountTransaction and wallet/getAccountTransactions
- adds confirmations to wallet/getAccountTransaction and wallet/getAccountTransactions

- orders fields consistently between endpoints

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[X] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

**Review by EvanJRichard — APPROVED** [review]
```
curl -d '{"hash": "2f5c7e1bf5aece4ffc2b85949736c9974a494e31cd1e724e44769a514d57e0f8"}' -X GET http://localhost:8021/wallet/getAccountTransaction
{"status":200,"data":{"account":"default","transaction":{"hash":"2f5c7e1bf5aece4ffc2b85949736c9974a494e31cd1e724e44769a514d57e0f8","fee":"2","blockHash":"0000000000028d26f59647523430f06771d7c647cdd923bc94123acb935a0d1a","blockSequence":53908,"notesCount":2,"spendsCount":4,"mintsCount":0,"burnsCount":0,"expiration":53921,"timestamp":1681146498143,"submittedSequence":53908,"assetBalanceDeltas":[{"assetId":"d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6","assetName":"2449524f4e","delta":"50000"}],"notes":[{"isOwner":true,"owner":"e0ed4dc14e6591595da1bdaf5f8acb7c01bc8687755bb71e59ea05264e2e5db7","memo":"","value":"50000","assetId":"d7c86706f5817aa718cd1cfad03233bcd64a7789fd9422d3b17af6823a7e6ac6","assetName":"2449524f4e","sender":"89f22093001943e7885ef9ead1c5b05a9fc6cdf7d36fd5b070b328555fba20ce","spent":false}],"status":"confirmed","type":"receive","confirmations":2}}}
```

File: ironfish/src/rpc/routes/wallet/getTransaction.ts [hunk 1]
```diff
@@ -118,8 +122,10 @@ router.register<typeof GetAccountTransactionRequestSchema, GetAccountTransaction
 
     const notes = await getAccountDecryptedNotes(node, account, transaction)
 
+    const confirmations = request.data.confirmations ?? node.config.get('confirmations')
```
Comment (NullSoldier) [inline ironfish/src/rpc/routes/wallet/getTransaction.ts:125]: getTransactonStatus already materializes this so I dont think we need this code in the RPC do we?

Comment (hughy) [reply ironfish/src/rpc/routes/wallet/getTransaction.ts:125]: I'm materializing it here only to include it in the response

File: ironfish/src/rpc/routes/wallet/getTransactions.ts [hunk 2]
```diff
@@ -106,7 +114,7 @@ router.register<typeof GetAccountTransactionsRequestSchema, GetAccountTransactio
 
     const options = {
       headSequence,
-      confirmations: request.data.confirmations,
+      confirmations: request.data.confirmations ?? node.config.get('confirmations'),
```
[the remark this answers is not in the document]
Comment (hughy) [reply ironfish/src/rpc/routes/wallet/getTransactions.ts:117]: Same as above -- we only need to materialize this here to include it in the response

**Review by NullSoldier — APPROVED** [review]

**Pull request merged**